src/fs/root/tree.ts

```diff
@@ -246,6 +275,50 @@ export default class RootTree implements Puttable {
   }
 
 
+  // SHARING
+  // -------
+
+  async addShares(links: SimpleLink[]): Promise<this> {
+    this.sharedLinks = links.reduce(
+      (acc, link) => ({ ...acc, [link.name]: link }),
+      this.sharedLinks
+    )
+
+    const cid = await ipfs.add(
```

Comment:
This doesn't actually work, because these links will not be recursively pinned.
This should use `ipfs.dag.put` instead.